src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -13,318 +13,332 @@ import IndexPatternsFieldListProvider from 'ui/index_patterns/_field_list';
       getIds.clearCache();
       return body;
-    };
-
-    function setId(id) {
-      return self.id = id;
     }
 
-    self.create = function () {
-      let body = self.prepBody();
-      return docSource.doCreate(body)
-      .then(setId)
-      .catch(function (err) {
-        if (_.get(err, 'origError.status') === 409) {
-          let confirmMessage = 'Are you sure you want to overwrite this?';
-
-          return safeConfirm(confirmMessage).then(
-            function () {
-              return Promise.try(function () {
-                const cached = patternCache.get(self.id);
+    create() {
+      const body = this.prepBody();
+      return this._docSource
+        .doCreate(body)
+        .then(id => this._setId(id))
+        .catch(err => {
+          if (_.get(err, 'origError.status') !== 409) {
+            return Promise.resolve(false);
+          }
+          const confirmMessage = 'Are you sure you want to overwrite this?';
+
+          return safeConfirm(confirmMessage)
+            .then(() => Promise
+              .try(() => {
+                const cached = patternCache.get(this.id);
                 if (cached) {
                   return cached.then(pattern => pattern.destroy());
                 }
               })
-              .then(() => docSource.doIndex(body))
-              .then(setId);
-            },
-            _.constant(false) // if the user doesn't overwrite, resolve with false
-          );
-        }
-        return Promise.resolve(false);
-      });
-    };
-
-    self.save = function () {
-      let body = self.prepBody();
-      return docSource.doIndex(body).then(setId);
-    };
-
-    self.refreshFields = function () {
-      return mapper.clearCache(self)
-      .then(self._fetchFields)
-      .then(self.save);
-    };
-
-    self._fetchFields = function () {
-      return mapper.getFieldsForIndexPattern(self, true)
-      .then(function (fields) {
-        // append existing scripted fields
-        fields = fields.concat(self.getScriptedFields());
-
-        // initialize self.field with this field list
-        initFields(fields);
-      });
-    };
+              .then(() => this._docSource.doIndex(body))
+              .then(id => this._setId(id)),
+              _.constant(false) // if the user doesn't overwrite, resolve with false
+            );
+        });
+    }
 
-    self.toJSON = function () {
-      return self.id;
-    };
+    save() {
+      const body = this.prepBody();
+      return this._docSource
+        .doIndex(body)
+        .then(id => this._setId(id));
+    }
 
-    self.toString = function () {
-      return '' + self.toJSON();
-    };
+    refreshFields() {
+      return mapper
+        .clearCache(this)
+        .then(() => this._fetchFields())
+        .then(() => this.save());
+    }
 
-    self.destroy = function () {
-      patternCache.clear(self.id);
-      docSource.destroy();
-    };
+    _initFields(fields) {
+      this._initialized = true;
```

Comment:
Feels like this would coorelate to `this.init()`, not `this._initFields()`. Maybe checking `this.fields` is sufficient, or rename to `this._fieldsInitialized`